src/controllers/auth.js

```diff
@@ -190,9 +192,18 @@ api.loginSocial = function(req, res, next) {
 
 /**
  * DELETE /user/auth/social
- * TODO implement
  */
-api.deleteSocial = function(req,res,next){next()}
+api.deleteSocial = function(req,res,next){
+  if (!res.locals.user.auth.local.username)
+    return res.json(401, {err:"Account lacks another authentication method, can't detach Facebook"});
+  //FIXME for some reason, the following gives https://gist.github.com/lefnire/f93eb306069b9089d123
```

Comment:
dayStart will likely be referring to `preferences.dayStart` (custom day
start) from the user's object, so `preferences` isn't available at this
point, or the entire user object isn't. Apologies if you knew that and were
wanting significantly more useful information. :)

On 4 February 2015 at 13:49, Blade Barringer <notifications@github.com>
wrote:

> In src/controllers/auth.js
> <https://github.com/HabitRPG/habitrpg/pull/4625#discussion_r24061193>:
>
> >   */
> > -api.deleteSocial = function(req,res,next){next()}
> > +api.deleteSocial = function(req,res,next){
> > +  if (!res.locals.user.auth.local.username)
> > +    return res.json(401, {err:"Account lacks another authentication method, can't detach Facebook"});
> > +  //FIXME for some reason, the following gives https://gist.github.com/lefnire/f93eb306069b9089d123
>
> Weird. What do you think this is about?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HabitRPG/habitrpg/pull/4625/files#r24061193>.
>